nvidia-apply-extra.c

```diff
@@ -265,6 +265,60 @@ find_line_offset (int fd, int skip_lines)
   return 0;
 }
 
+static void
+replace_string_in_file (const char *path,
+                        const char *string,
+                        const char *replacement)
+{
+  char *buffer, *p;
+  size_t len;
+  long fsize;
+  FILE *f;
+
+  if ((f = fopen (path, "r+")) == NULL)
+    die_with_error ("reading file %s", path);
+
+  fseek (f, 0, SEEK_END);
+  fsize = ftell (f);
+  fseek (f, 0, SEEK_SET);
+
+  if ((buffer = malloc (fsize + 1)) == NULL)
+    die ("out of memory");
+
+  if ((len = fread (buffer, 1, fsize, f)) != fsize)
+    die ("failed to read file %s", path);
```

Comment:
Ideally this should handle short reads and do multiple reads.

Reply:
This was very much a `"We only have one usage for it today and its a small file so its fine"`.